tasks/rh-sign-image/rh-sign-image.yaml

```diff
@@ -112,58 +165,70 @@ spec:
               for tag in ${TAGS}; do
                 # Iterate over both rh-registry-repo and registry-access-repo
                 for registry_reference in ${rh_registry_repo} ${registry_access_repo}; do
-                  echo "Creating InternalRequest to sign image with tag ${tag}:"
+                  echo "Creating ${requestType} to sign image with tag ${tag}:"
                   echo "- reference=${registry_reference}:${tag}"
                   echo "- manifest_digest=${manifest_digest}"
                   echo "- requester=$(params.requester)"
 
-                  internal-request -r "${request}" \
-                      -p pipeline_image=${pipeline_image} \
-                      -p reference=${registry_reference}:${tag} \
-                      -p manifest_digest=${manifest_digest} \
-                      -p requester=$(params.requester) \
-                      -p config_map_name=${config_map_name} \
-                      -l ${TASK_LABEL}=${TASK_ID} \
-                      -l ${PIPELINERUN_LABEL}=$(params.pipelineRunUid) \
-                      -s false
-                  ((++count))
-
-                  if [ "$count" -eq "$N" ]; then
-                      wait-for-ir -l ${TASK_LABEL}=${TASK_ID} -t $(params.requestTimeout)
-                      count=0
-                  fi
+                  while (( ${RUNNING_JOBS@P} >= "$CONCURRENT_LIMIT" )); do
+                    wait -n
+                  done
+
+                  ${requestType} -r "${request}" \
+                    -p pipeline_image="${pipeline_image}" \
+                    -p reference="${registry_reference}:${tag}" \
+                    -p manifest_digest="${manifest_digest}" \
+                    -p requester="$(params.requester)" \
+                    -p config_map_name="${config_map_name}" \
+                    -p taskGitUrl="$(params.taskGitUrl)" \
+                    -p taskGitRevision="$(params.taskGitRevision)" \
+                    -l ${TASK_LABEL}="${TASK_ID}" \
+                    -l ${PIPELINERUN_LABEL}="$(params.pipelineRunUid)" \
+                    ${taskGitUrlArg:+$taskGitUrlArg} ${taskGitRevisionArg:+$taskGitRevisionArg} \
+                    ${serviceAccountArg:+$serviceAccountArg} -s true &
                 done
               done
             done
 
+            # Wait for remaining processes to finish
+            while (( ${RUNNING_JOBS@P} > 0 )); do
+                wait -n
```

Comment:
Just FYI, I was suggesting to use `sleep` here, because I thought `wait -n` would wait for the next job in order they were started, but Johnny proved me wrong, so `wait -n` is fine: https://github.com/konflux-ci/release-service-catalog/pull/575#discussion_r1783164220

Reply:
can we resolve this now?